fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -188,7 +188,8 @@ public RecordQueryPlan plan(@Nonnull RecordQuery query) {
 
         final PlanContext planContext = getPlanContext(query);
 
-        final QueryComponent filter = BooleanNormalizer.normalize(query.getFilter());
+        final BooleanNormalizer normalizer = BooleanNormalizer.withLimit(complexityThreshold);
```

Comment:
Are the complexity threshold and the DNF size limit similar enough that they should share the same constant, or should it be something that can be set separately?

I suppose that if we want to change this later, we probably can, though it would be somewhat API breaking under the strictest of definitions.

Reply:
Well, I wrote down my justification above:
> How the size limit gets passed to the normalizer. I used the existing plan complexity limit that we already pass to the planner. The argument here is that we mostly normalize so we can produce union plans and so there's no point in producing a DNF that would produce a union that's too big to pass the complexity limit. I don't think this especially needs to be configurable since it's very much a planner implementation detail.

Maybe it's more correct to shrink the limit a little more but I don't think it's important enough to matter. Regarding the "API breaking", taking that definition of API breaking means that any change to the planner that affects what plans get produced is "API breaking". I don't think that's a useful definition and I honestly wouldn't worry about it.

Reply:
Okay, I can live with it as is (at least as a version 1).